_posts/2020-05-06-cockpit-ansible.md

```diff
@@ -0,0 +1,28 @@
+---
+title: Scale Cockpit troubleshooting to your computing fleet
+author: pitti
+date: 2020-05-06
+category: announcement
+tags: cockpit selinux ansible scaling troubleshooting
+comments: true
+---
+
+You know Cockpit as a troubleshooting and learning tool for an individual machine. But once you discovered and tested a solution, wouldn't it be nice to apply it to all your other machines in your data center? Of course not every problem works that way: you don't extend an LVM with a new hard disk on a hundred machines at the same time. But there *are* situations where this makes sense, for example adjusting the [SELinux policy](https://www.redhat.com/en/topics/linux/what-is-selinux) to your needs.
+
+[Cockpit 210](./cockpit-210.html) introduced a new approach that we want to explore: "Show me what I have done". If you apply SELinux policy changes, you can now see the changes compared to the Operating System defaults in human readable form, as well as a shell script or [Ansible](https://www.ansible.com/) [role](https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse) that is suitable for automation:
+
+![SElinux autoscript ansible](/images/selinux-autoscript-ansible.png)
+
+Have a look at this 3.5 minute demo that shows the whole workflow from "I have a problem that I need to debug" over "I figure out the solution with Cockpit" to "I apply the tested solution to my entire fleet":
+
+<iframe width="960" height="720" src="//youtube.com/embed/ChXoNofPIjw?rel=0" frameborder="0" allowfullscreen></iframe>
+
+Not every page in Cockpit is amenable to this approach. First and foremost, Cockpit always shows the server's current *state*, not configuration. In a lot of cases it is not well-defined whether something that you see is configuration, or dynamic state (like the IP address of a DHCP network card) or even hardware properties (like the number, capacity, and serials of hard disks).
```

Comment:
```suggestion
While this approach works for SELinux, not every part of Cockpit can be used in a manner like this. First and foremost, Cockpit always shows the server's current *state*, not its configuration. In a lot of cases, the difference between static configurations, dynamic state (like the IP address of a DHCP network card), or even hardware properties (like the number, capacity, and serials of hard disks) are not necessarily obvious at a glance.
```

Reply:
"configurations" sounds strange here, I think the singular is better.